lua/hover/providers/lsp.lua

```diff
@@ -6,26 +6,35 @@ require('hover').register {
   name = 'LSP',
   priority = 1000,
   enabled = function()
-    for _, client in pairs(get_clients()) do
+    for _, client in pairs(get_clients({ bufnr = 0 })) do
       if client and client.supports_method('textDocument/hover') then
         return true
       end
     end
     return false
   end,
   execute = function(opts, done)
+    local util = require('vim.lsp.util')
+
+    local row, col = unpack(opts.pos)
+    local offset_encoding = util._get_offset_encoding(opts.bufnr)
+    row = row - 1
+    local line = vim.api.nvim_buf_get_lines(opts.bufnr, row, row + 1, true)[1]
+    if line then
+      col = util._str_utfindex_enc(line, col, offset_encoding)
+    end
+
     local params = {
-      textDocument = { uri = vim.uri_from_bufnr(opts.bufnr) },
+      textDocument = util.make_text_document_params(opts.bufnr),
       position = {
-        line = opts.pos[1],
-        character = opts.pos[2]
+        line = row,
+        character = col,
```

Comment:
This only considers offset_encoding for a one client and won't be correct for multiple.